### PR TITLE
Tweaks on the individual form and herd view

### LINKED ISSUE #349: Individual number validation issues
- [x] Fix validation of herd numbers. Currently the messages lie about the origin of the mother and father
- [x] Revamp the herd text field: change it to dropdown. Make a new field for the individual number.

### PR DESCRIPTION
**Pull request by HenrikeW**
I didn't rebase on the develop branch with the new layout but it doesn't seem to be a problem for merging.

Changes in the individual form:
- you now choose the origin herd with a dropdown and it auto-fills the first part of the individual's number
- as soon as you choose the birth date, even the first 2 of the individual's last 4 digits will show up, in line with the rabbit peoples' system for creating the individual numbers: "year, litter, individual". This hopefully helps the user to not make mistakes.
- I changed the order of input fields, mostly for design reasons.
- the validation doesn't lie about the herds any more :) I hadn't thought about that not all herds have a herd.name and my if-statement was too strict with this.

Changes in the herd view:
- the list of breeding events has moved to its own tab.
- For this, I also introduced the prop `full` to the `popup` in `message_context.tsx` to prevent the popup from changing size when switching between tabs.

### CODE REVIEW

**Comment by airenzp** [comment]
I wanted to try these changes with the new layout so I merged it with the develop branch and added my fixes

**Comment by HenrikeW** [comment]
I added some more things: 
- choose mother and father first, then origin herd options are limited to two. Adjusted layout to this, even made it (sort of) responsive
- added some extra validation for the individual numbers format

File: frontend/src/data_context_global.tsx [hunk 1]
```diff
@@ -121,6 +121,20 @@ export interface Breeding {
   litter_size: number | null;
 }
 
+export interface ExtendedBreeding {
+  [key: string]: any;
+  id?: number;
+  breed_date: string | null;
+  breed_notes: string;
```
Comment (pontus) [inline frontend/src/data_context_global.tsx]: Are notes mandatory?

**Review by airenzp — APPROVED** [review]
I have been testing this PR and I am quite satisfied with the result.

File: frontend/src/data_context_global.tsx (hunk 1 above)
Comment (jonandernovella) [reply frontend/src/data_context_global.tsx]: The mandatory / non-mandatory fields should be reworked. At the moment it's quite messy...

File: frontend/src/individual_add.tsx [hunk 2]
```diff
@@ -449,7 +466,9 @@ export function IndividualAdd({
   const resetSibling = () => {
     const numberParts: string[] = individual?.number?.split("-");
     const sibling: Individual = {
-      number: numberParts ? numberParts[0] + "-" : null,
+      number: numberParts
+        ? numberParts[0] + "-" + numberParts[1][0] + numberParts[1][1]
```
Comment (pontus) [inline frontend/src/individual_add.tsx]: If I undertand recent discussions (https://nbisweden.slack.com/archives/C010DDHHZ4N/p1628502747044800?thread_ts=1628500448.031700&cid=C010DDHHZ4N), the this should be numberParts[0] dash the first three digits from numberParts[1]?

File: frontend/src/data_context_global.tsx [hunk 3]
```diff
@@ -121,6 +121,20 @@ export interface Breeding {
   litter_size: number | null;
 }
 
+export interface ExtendedBreeding {
```
Comment (pontus) [inline frontend/src/data_context_global.tsx]: Should probably extend `Breeding`? (https://www.typescriptlang.org/docs/handbook/2/objects.html#extending-types)

**Review by pontus — APPROVED** [review]
Don't see any obvious issues.

There are a few things I note, but these are old sins and I'm not sure I have the full view anyway (e.g. strings for dates vs `Date` and so on).

File: frontend/src/data_context_global.tsx (hunk 1 above)
Comment (HenrikeW) [reply frontend/src/data_context_global.tsx]: yes, I agree. But I'm pretty sure the breed_notes shouldn't be mandatory, so I'll change it.

File: frontend/src/individual_add.tsx (hunk 2 above)
Comment (HenrikeW) [reply frontend/src/individual_add.tsx]: good catch! Yes, I think you're right. I'll change it.

File: frontend/src/data_context_global.tsx (hunk 3 above)
Comment (HenrikeW) [reply frontend/src/data_context_global.tsx]: Yes, indeed. ExtendedBreeding should extend breeding. I guess I had a thought when writing this and then didn't fully think it to the end :)

**Review by jonandernovella — APPROVED** [review]

**Pull request merged**